src/Core/Algorithms/Legacy/Fields/Cleanup/ReorderNormalCoherentlyAlgo.cc

```diff
@@ -0,0 +1,273 @@
+    
+    for(i=0;i<3;i++)
+      elem[elemID][i]=nodesFromFace[i];
+  }
+  
+  for(i=0;i<m;i++)
+    elemNeighbors.push_back(dummy);
+  
+  for(i=0;i<n;i++)
+    elemsOfVert.push_back(dummy);
+  
+  //get elements
+  for(i=0;i<m;i++)
+  {
+    for(j=0;j<3;j++)
+      elemsOfVert[elem[i][j]].insert(i);
+  }
+  
+  //get edges
+  //create graph
+  for(i=0;i<m;i++)
+  {
+    for(j=0;j<3;j++)
+    {
+      k = (j+1)%3;
+      std::set<int>::iterator it;
+      for (it = elemsOfVert[elem[i][j]].begin(); it != elemsOfVert[elem[i][j]].end(); it++)
+      {
+        if (elemsOfVert[elem[i][k]].find(*it) != elemsOfVert[elem[i][k]].end())
+        {
+          if (*it != i)
+          {
+            elemNeighbors[i].insert(*it);
+            elemNeighbors[i].insert(*it);
+          }
+        }
+      }
+    }
+  }
+    
+    //traverse graph
+    std::vector<bool> elemTraversed;
+    for(i=0;i<m;i++)
+      elemTraversed.push_back(false);
+    
+    std::queue<int> bfs;
+    bfs.push(0);
+    elemTraversed[0]=true;
+    std::set<std::pair<int,int>> edges;
+    
+  while(!bfs.empty())
+  {
+    //insert elems not traversed
+    std::set<int>::iterator it;
+    i=bfs.front();
+    
+    //invert in the output field
+    VMesh::Face::index_type elemID=i;
+    inputVMesh->get_nodes(nodesFromFace, elemID);
+    
+    for(it=elemNeighbors[i].begin();it!=elemNeighbors[i].end();it++)
+    {
+      if(!elemTraversed[*it])
+      {
+        bfs.push(*it);
+        elemTraversed[*it]=true;
+      }
+    }
+    //reordered elem i if desired
+    bool flag=false;
+    for (j = 0; j < 3 && flag == false; j++)
+    {
+      k = (j+1)%3;
+      if (edges.find(std::make_pair(elem[i][j], elem[i][k])) != edges.end())
+      {
+        invertedElements.push_back(i);
+        // invert ordering
+        flag = true;
+        int temp = elem[i][j];
+        elem[i][j]=elem[i][k];
+        elem[i][k]=temp;
+        nodesFromFace[j] = elem[i][j];
+        nodesFromFace[k] = elem[i][k];
+      }
+    }
+    for (j = 0; j < 3; j++)
+    {
+      k=(j+1)%3;
+      edges.insert(std::make_pair(elem[i][j], elem[i][k]));
+    }
+    outputVMesh->add_elem(nodesFromFace);
+    
+    bfs.pop();
+  }
+  
+  VField* outputVField=outputField->vfield();
+  outputVField->resize_values();
+  std::vector<VMesh::index_type> order;
+  
+    // Copy field data (non-linear not supoorted, check made upstream)
+  int basis_order=inputVField->basis_order();
+  if (basis_order == 0)
+  {
+    VField::size_type size = order.size();
+    for(VField::index_type idx = 0; idx < size; idx++)
+      outputVField->copy_value(inputVField, order[idx], idx);
+  }
+  else if (basis_order == 1)
+  {
+    outputVField->copy_values(inputVField);
+  }
+  
+  CopyProperties(*inputField,*outputField);
+
+  
+  if (invertedElements.size() > 0)
+  {
+    std::ostringstream oss;
+    oss << invertedElements.size() << "elements were found to be incorrectly oriented in input TriSurf mesh.";
+    warning(oss.str());
```

Comment:
we should change this to a remark

Reply:
Do you refer to TriSurf in the warning?

Reply:
line 239.  it's warning about the number of elements flipped, but that is the job of the module, so it should be a remark

Reply:
I see.